src/compiler/preprocess/index.ts

```diff
@@ -83,6 +83,82 @@ async function replace_async(
 	return out.concat(final_content);
 }
 
+
+
+/**
+ * import decoded sourcemap from mozilla/source-map/SourceMapGenerator
+ * forked from source-map/lib/source-map-generator.js
+ * from methods _serializeMappings and toJSON
+ */
+function decodedSourcemapFromSourceMapGenerator(this: any) {
```

Comment:
can the type here be `SourceMapGenerator` instead of `any`?

Reply:
Only if we can write the import as `import type { .. } from ..`, else I fear `source-map` might end up in the bundle.

Reply:
here come the typescript fans : P
IMO that makes zero sense here, cos its an internal function
(deja vu ....)

Reply:
True, all the `_`-functions are very likely not typed in the public api interface.

Reply:
there is also a technical argument for not using the official type:
the import function is using "hidden" properties, that are not defined in [source-map.d.ts](https://github.com/mozilla/source-map/blob/7a263408de1b60e60b6220ca3e36d9b3e4aa047d/source-map.d.ts#L286)

Reply:
Even if it's an internal function it still makes sense to me to add the type. It helps with readability and makes it much easier to edit this function in VS Code by giving the IDE more info about what this type actually is

But given that we're accessing a lot of `_` prefixed private variables, you're right that it seems not worth it in this case